docs/serdes.rst

```diff
@@ -2,17 +2,41 @@
 Serialization/Deserialization protocol
 ======================================
 
-Serializer and Deserializer are used to marshal and unmarshal messages according to a given schema by adding schema version information along with the payload.
-There are different protocols in how schema version is passed along with the payload. This framework allows to build custom ser/des protocols by users.
+Serializer and Deserializer are used to marshal and unmarshal messages according to a given schema by adding schema version
+information either in the Message Header [or] along with the payload. There are different protocols and/or formats in
+how schema version are added with the Message. This framework allows to build custom ser/des protocols by users.
+
+Message Formats
+```````````````
+
+In Kafka v0.11.0.0, Record Header support is introduced. With this support, user can optionally save the schema
+information in the header rather than prepending it in the body of the message. (See Message Format 2)
+
+Message Format 1
+""""""""""""""""
+|header:| |body:<version_info><payload>|
+Prepends the schema version information along with the user payload in the body of the Message.
```

Comment:
We may want to add this format is the default format to maintain backward compatibility. It is good to mention how users can use message format 2. Incase of Kafka serializer/deserializer, add details on how to use headers based schema version serialization [here]( https://github.com/hortonworks/registry/blob/master/docs/schema-registry.rst#kafka-producer-integration-with-schemaregistry)

Reply:
Addressed your comments. Please take another look.